extensions/yaml-configuration/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/FileSystemApplicationYamlConfigSource.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import java.io.FileNotFoundException;
+import java.io.IOError;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FileSystemApplicationYamlConfigSource extends YamlConfigSource {
+
+    private static final long serialVersionUID = -3881751839186883952L;
+
+    private static final String YAML_FILE = "application.yml";
+    private static final int PRIORITY = 280;
```

Comment:
The properties-based version of this config source priority is `260`. See `ApplicationPropertiesConfigSource$InFileSystem`.